src/error.rs

```diff
@@ -100,6 +109,9 @@ pub enum APIError {
     #[error("Invalid pubkey")]
     InvalidPubkey,
 
+    #[error("Invalid swap string: {0}")]
+    InvalidSwapString(String),
```

Comment:
```suggestion
    #[error("Invalid swap string '{0}': {1}")]
    InvalidSwapString(String, String),
```

the second parameter will be the reason of the string being invalid. therefore, when mapping the error from `SwapString::from_str` please put the returned error string instead of ignoring it